tools-v2/pkg/cli/command/curvebs/list/dir/dir.go

```diff
@@ -0,0 +1,224 @@
+	"log"
+	"time"
+)
+
+const (
+	dirExample = `$ curve bs list dir --dir /`
+)
+
+type ListDirRpc struct {
+	Info          *basecmd.Rpc
+	Request       *nameserver2.ListDirRequest
+	curveFSClient nameserver2.CurveFSServiceClient
+}
+
+var _ basecmd.RpcFunc = (*ListDirRpc)(nil) // check interface
+
+type DirCommand struct {
+	basecmd.FinalCurveCmd
+	Rpc []*ListDirRpc
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*DirCommand)(nil) // check interface
+
+func (lRpc *ListDirRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	lRpc.curveFSClient = nameserver2.NewCurveFSServiceClient(cc)
+}
+
+func (lRpc *ListDirRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return lRpc.curveFSClient.ListDir(ctx, lRpc.Request)
+}
+
+func NewDirCommand() *cobra.Command {
+	return NewListDirCommand().Cmd
+}
+
+func NewListDirCommand() *DirCommand {
+	lsCmd := &DirCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "dir",
+			Short:   "list directory information in curvebs",
+			Example: dirExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&lsCmd.FinalCurveCmd, lsCmd)
+	return lsCmd
+}
+
+// AddFlags implements basecmd.FinalCurveCmdFunc
+func (pCmd *DirCommand) AddFlags() {
+	config.AddBsMdsFlagOption(pCmd.Cmd)
+	config.AddRpcRetryTimesFlag(pCmd.Cmd)
+	config.AddRpcTimeoutFlag(pCmd.Cmd)
+	config.AddBsDirOptionFlag(pCmd.Cmd)
+	config.AddBsUserOptionFlag(pCmd.Cmd)
+	config.AddBsPasswordOptionFlag(pCmd.Cmd)
+}
+
+// Init implements basecmd.FinalCurveCmdFunc
+func (pCmd *DirCommand) Init(cmd *cobra.Command, args []string) error {
+	mdsAddrs, err := config.GetBsMdsAddrSlice(pCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	timeout := config.GetFlagDuration(pCmd.Cmd, config.RPCTIMEOUT)
+	retrytimes := config.GetFlagInt32(pCmd.Cmd, config.RPCRETRYTIMES)
+	fileName := config.GetBsFlagString(pCmd.Cmd, config.CURVEBS_DIR)
+	owner := config.GetBsFlagString(pCmd.Cmd, config.CURVEBS_USER)
+	date, errDat := cobrautil.GetTimeofDayUs()
+	if errDat.TypeCode() != cmderror.CODE_SUCCESS {
+		return errDat.ToError()
+	}
+
+	rpc := &ListDirRpc{
+		Request: &nameserver2.ListDirRequest{
+			FileName: &fileName,
+			Owner:    &owner,
+			Date:     &date,
+		},
+		Info: basecmd.NewRpc(mdsAddrs, timeout, retrytimes, "ListDir"),
+	}
+	// auth
+	password := config.GetBsFlagString(pCmd.Cmd, config.CURVEBS_PASSWORD)
+	if owner == viper.GetString(config.VIPER_CURVEBS_USER) && len(password) != 0 {
+		strSig := cobrautil.GetString2Signature(date, owner)
+		sig := cobrautil.CalcString2Signature(strSig, password)
+		rpc.Request.Signature = &sig
+	}
+	pCmd.Rpc = append(pCmd.Rpc, rpc)
+	header := []string{
+		cobrautil.ROW_ID,
+		cobrautil.ROW_FILE_NAME,
+		cobrautil.ROW_PARENT_ID,
+		cobrautil.ROW_FILE_TYPE,
+		cobrautil.ROW_OWNER,
+		cobrautil.ROW_CTIME,
+		cobrautil.ROW_ALLOC_SIZE,
+		cobrautil.ROW_FILE_SIZE,
+	}
+	pCmd.SetHeader(header)
+	pCmd.TableNew.SetAutoMergeCellsByColumnIndex(cobrautil.GetIndexSlice(
+		pCmd.Header, header,
```

Comment:
only needs to be merged,like:
cobrautil.ROW_OWNER、cobrautil.ROW_FILE_TYPE、cobrautil.ROW_PARENT_ID.
IMO,there is no need to merge Ctime, alloc zie, and size